Cocoapod Spec?

Cocoapod distribution will reduce the barrier for some of us to try out swiftz.  Is this in swiftz future?

I'm not very familiar with cocoapods (or swift for that matter), but I went ahead and tried to create a podspec file. The file I created looks like this: 

```ruby
Pod::Spec.new do |s|
  s.name        = "Swiftz"
  s.version     = "0.1.4"
  s.summary     = "Swiftz is a Swift library for functional programming."
  s.homepage    = "https://github.com/typelift/Swiftz"
  s.license     = { :type => "BSD" }
  s.authors     = { "CodaFi" => "devteam.codafi@gmail.com"}

  s.requires_arc = true
  s.osx.deployment_target = "10.9"
  s.ios.deployment_target = "8.0"
  s.source   = { :git => "https://github.com/typelift/Swiftz.git", :tag => "v0.1.4"}
  s.source_files = "Swiftz/*.swift"
end
```

As you can see, it's set up to grab the `v0.1.4` tag from the repo. 

Then I ran `pod spec lint` which grabs the tag from the repo and runs the linter. I got the following errors:

```bash
➜  Swiftz git:(cocoapod) ✗ pod spec lint

 -> Swiftz (0.1.4)
    - ERROR | Returned an unsuccessful exit code. You can use `--verbose` for more information.
    - WARN  |  Swiftz/Swiftz/ArrayExt.swift:89:34: warning: extraneous '#' in parameter: 'rhs' is already the keyword argument name
    - ERROR |  Swiftz/Swiftz/Arrow.swift:68:8: error: class methods are only allowed within classes; use 'static' to declare a static method
    - ERROR |  Swiftz/Swiftz/Arrow.swift:98:8: error: class methods are only allowed within classes; use 'static' to declare a static method
    - ERROR |  Swiftz/Swiftz/Arrow.swift:173:8: error: class methods are only allowed within classes; use 'static' to declare a static method
    - ERROR |  Swiftz/Swiftz/Arrow.swift:190:8: error: class methods are only allowed within classes; use 'static' to declare a static method
    - ERROR |  Swiftz/Swiftz/Bounded.swift:14:8: error: class methods are only allowed within classes; use 'static' to declare a static method
    - ERROR |  Swiftz/Swiftz/Bounded.swift:15:8: error: class methods are only allowed within classes; use 'static' to declare a static method
    - ERROR |  Swiftz/Swiftz/Category.swift:33:8: error: class methods are only allowed within classes; use 'static' to declare a static method
    - ERROR |  Swiftz/Swiftz/Functor.swift:30:19: error: 'autoclosure' attribute is now an attribute of the parameter declaration, not its type
    - ERROR |  Swiftz/Swiftz/Functor.swift:69:19: error: 'autoclosure' attribute is now an attribute of the parameter declaration, not its type
    - ERROR |  Swiftz/Swiftz/Future.swift:37:43: error: 'autoclosure' attribute is now an attribute of the parameter declaration, not its type
    - ERROR |  Swiftz/Swiftz/HList.swift:16:8: error: class methods are only allowed within classes; use 'static' to declare a static method
    - ERROR |  Swiftz/Swiftz/HList.swift:17:8: error: class methods are only allowed within classes; use 'static' to declare a static method
    - ERROR |  Swiftz/Swiftz/HList.swift:18:8: error: class methods are only allowed within classes; use 'static' to declare a static method
    - ERROR |  Swiftz/Swiftz/HList.swift:19:8: error: class methods are only allowed within classes; use 'static' to declare a static method
    - ERROR |  Swiftz/Swiftz/JSON.swift:198:8: error: class methods are only allowed within classes; use 'static' to declare a static method
    - ERROR |  Swiftz/Swiftz/JSON.swift:203:8: error: class methods are only allowed within classes; use 'static' to declare a static method
    - ERROR |  Swiftz/Swiftz/List.swift:38:16: error: 'autoclosure' attribute is now an attribute of the parameter declaration, not its type
    - ERROR |  Swiftz/Swiftz/MVar.swift:40:29: error: 'autoclosure' attribute is now an attribute of the parameter declaration, not its type
    - ERROR |  Swiftz/Swiftz/Monoid.swift:12:8: error: class properties are only allowed within classes; use 'static' to declare a static property
    - ERROR |  Swiftz/Swiftz/Monoid.swift:23:20: error: 'autoclosure' attribute is now an attribute of the parameter declaration, not its type
    - ERROR |  Swiftz/Swiftz/Monoid.swift:40:20: error: 'autoclosure' attribute is now an attribute of the parameter declaration, not its type
    - ERROR |  Swiftz/Swiftz/Monoid.swift:57:20: error: 'autoclosure' attribute is now an attribute of the parameter declaration, not its type
    - ERROR |  Swiftz/Swiftz/Monoid.swift:82:20: error: 'autoclosure' attribute is now an attribute of the parameter declaration, not its type
    - ERROR |  Swiftz/Swiftz/Monoid.swift:103:20: error: 'autoclosure' attribute is now an attribute of the parameter declaration, not its type
    - ERROR |  Swiftz/Swiftz/Num.swift:13:8: error: class properties are only allowed within classes; use 'static' to declare a static property
    - ERROR |  Swiftz/Swiftz/Num.swift:14:8: error: class properties are only allowed within classes; use 'static' to declare a static property
    - ERROR |  Swiftz/Swiftz/Pointed.swift:12:8: error: class methods are only allowed within classes; use 'static' to declare a static method
    - ERROR |  Swiftz/Swiftz/SYB.swift:15:8: error: class methods are only allowed within classes; use 'static' to declare a static method
    - ERROR |  Swiftz/Swiftz/SYB.swift:17:8: error: class methods are only allowed within classes; use 'static' to declare a static method
    - ERROR |  Swiftz/Swiftz/Semigroup.swift:28:19: error: 'autoclosure' attribute is now an attribute of the parameter declaration, not its type
    - ERROR |  Swiftz/Swiftz/Semigroup.swift:45:19: error: 'autoclosure' attribute is now an attribute of the parameter declaration, not its type
    - ERROR |  Swiftz/Swiftz/Comonad.swift:23:11: error: use of undeclared type 'Box'
    - ERROR |  Swiftz/Swiftz/Comonad.swift:34:11: error: use of undeclared type 'Box'
    - ERROR |  Swiftz/Swiftz/Comonad.swift:40:11: error: use of undeclared type 'Box'
    - ERROR |  Swiftz/Swiftz/Pointed.swift:15:11: error: use of undeclared type 'Box'
    - ERROR |  Swiftz/Swiftz/Pointed.swift:21:11: error: use of undeclared type 'Result'
    - ERROR |  Swiftz/Swiftz/Pointed.swift:29:11: error: use of undeclared type 'Either'
    - ERROR |  Swiftz/Swiftz/Those.swift:12:12: error: use of undeclared type 'Box'
    - ERROR |  Swiftz/Swiftz/Those.swift:13:12: error: use of undeclared type 'Box'
    - ERROR |  Swiftz/Swiftz/Those.swift:14:13: error: use of undeclared type 'Box'
    - ERROR |  Swiftz/Swiftz/NonEmptyList.swift:13:20: error: use of undeclared type 'Box'
    - ERROR |  Swiftz/Swiftz/ArrayZipper.swift:46:25: error: binary operator '<^>' cannot be applied to operands of type 'A -> B' and '[A]'
    - ERROR |  Swiftz/Swiftz/Arrow.swift:135:22: error: use of undeclared type 'Either'
    - ERROR |  Swiftz/Swiftz/Arrow.swift:137:23: error: use of undeclared type 'Either'
    - ERROR |  Swiftz/Swiftz/Arrow.swift:140:23: error: use of undeclared type 'Either'
    - ERROR |  Swiftz/Swiftz/Arrow.swift:145:23: error: use of undeclared type 'Either'
    - ERROR |  Swiftz/Swiftz/Bifunctor.swift:44:19: error: use of undeclared type 'Either'
    - ERROR |  Swiftz/Swiftz/Bifunctor.swift:42:16: error: use of undeclared type 'Either'
    - ERROR |  Swiftz/Swiftz/Bifunctor.swift:48:57: error: use of undeclared type 'Either'
    - ERROR |  Swiftz/Swiftz/Bifunctor.swift:57:40: error: use of undeclared type 'Either'
    - ERROR |  Swiftz/Swiftz/Bifunctor.swift:61:41: error: use of undeclared type 'Either'
    - ERROR |  Swiftz/Swiftz/Bifunctor.swift:38:8: error: type 'EitherBF<A, C>' does not conform to protocol 'Bifunctor'
    - ERROR |  Swiftz/Swiftz/Bifunctor.swift:49:10: error: 'EitherBF<A, C>' does not have a member named 'e'
    - ERROR |  Swiftz/Swiftz/Bifunctor.swift:50:9: error: enum case pattern cannot match values of the non-enum type '<<error type>>'
    - ERROR |  Swiftz/Swiftz/Bifunctor.swift:52:9: error: enum case pattern cannot match values of the non-enum type '<<error type>>'
    - ERROR |  Swiftz/Swiftz/Bifunctor.swift:110:24: error: argument for generic parameter 'S' could not be inferred
    - NOTE  |  Swiftz/Swiftz/Iso.swift:32:13: note: in call to function 'identity'
    - ERROR |  Swiftz/Swiftz/Bifunctor.swift:114:21: error: argument for generic parameter 'S' could not be inferred
    - ERROR |  Swiftz/Swiftz/Chan.swift:25:9: error: cannot assign to 'mutex' in 'self'
    - ERROR |  Swiftz/Swiftz/Chan.swift:26:8: error: cannot assign to 'cond' in 'self'
    - ERROR |  Swiftz/Swiftz/Chan.swift:29:8: error: cannot assign to 'matt' in 'self'

Analyzed 1 podspec.

[!] The spec did not pass validation.
```

I'm not sure if I'm even going about this correctly, but wanted to report back my firsts efforts. Perhaps others have ideas about to proceed.

@CodaFi thanks. Ran it again against the latest from master and got different errors:

```bash
➜  Swiftz git:(cocoapod) ✗ pod spec lint

 -> Swiftz (0.2.1)
    - ERROR | Returned an unsuccessful exit code. You can use `--verbose` for more information.
    - ERROR |  Swiftz/Swiftz/Comonad.swift:23:11: error: use of undeclared type 'Box'
    - ERROR |  Swiftz/Swiftz/Comonad.swift:34:11: error: use of undeclared type 'Box'
    - ERROR |  Swiftz/Swiftz/Comonad.swift:40:11: error: use of undeclared type 'Box'
    - ERROR |  Swiftz/Swiftz/Pointed.swift:15:11: error: use of undeclared type 'Box'
    - ERROR |  Swiftz/Swiftz/Pointed.swift:21:11: error: use of undeclared type 'Result'
    - ERROR |  Swiftz/Swiftz/Pointed.swift:29:11: error: use of undeclared type 'Either'
    - ERROR |  Swiftz/Swiftz/Those.swift:12:12: error: use of undeclared type 'Box'
    - ERROR |  Swiftz/Swiftz/Those.swift:13:12: error: use of undeclared type 'Box'
    - ERROR |  Swiftz/Swiftz/Those.swift:14:13: error: use of undeclared type 'Box'
    - ERROR |  Swiftz/Swiftz/NonEmptyList.swift:13:20: error: use of undeclared type 'Box'
    - ERROR |  Swiftz/Swiftz/ArrayExt.swift:312:9: error: cannot invoke 'span' with an argument list of type '([A], (_) -> _)'
    - NOTE  |  Swiftz/Swiftz/ArrayExt.swift:312:13: note: expected an argument list of type '([A], (A -> Bool))'
    - ERROR |  Swiftz/Swiftz/ArrayZipper.swift:46:25: error: binary operator '<^>' cannot be applied to operands of type 'A -> B' and '[A]'
    - ERROR |  Swiftz/Swiftz/Arrow.swift:135:22: error: use of undeclared type 'Either'
    - ERROR |  Swiftz/Swiftz/Arrow.swift:137:23: error: use of undeclared type 'Either'
    - ERROR |  Swiftz/Swiftz/Arrow.swift:140:23: error: use of undeclared type 'Either'
    - ERROR |  Swiftz/Swiftz/Arrow.swift:145:23: error: use of undeclared type 'Either'
    - ERROR |  Swiftz/Swiftz/Bifunctor.swift:44:19: error: use of undeclared type 'Either'
    - ERROR |  Swiftz/Swiftz/Bifunctor.swift:42:16: error: use of undeclared type 'Either'
    - ERROR |  Swiftz/Swiftz/Bifunctor.swift:48:57: error: use of undeclared type 'Either'
    - ERROR |  Swiftz/Swiftz/Bifunctor.swift:57:40: error: use of undeclared type 'Either'
    - ERROR |  Swiftz/Swiftz/Bifunctor.swift:61:41: error: use of undeclared type 'Either'
    - ERROR |  Swiftz/Swiftz/Bifunctor.swift:38:8: error: type 'EitherBF<A, C>' does not conform to protocol 'Bifunctor'
    - ERROR |  Swiftz/Swiftz/Bifunctor.swift:49:10: error: 'EitherBF<A, C>' does not have a member named 'e'
    - ERROR |  Swiftz/Swiftz/Bifunctor.swift:50:9: error: enum case pattern cannot match values of the non-enum type '<<error type>>'
    - ERROR |  Swiftz/Swiftz/Bifunctor.swift:52:9: error: enum case pattern cannot match values of the non-enum type '<<error type>>'
    - ERROR |  Swiftz/Swiftz/Bifunctor.swift:110:24: error: argument for generic parameter 'S' could not be inferred
    - NOTE  |  Swiftz/Swiftz/Iso.swift:35:13: note: in call to function 'identity'
    - ERROR |  Swiftz/Swiftz/Bifunctor.swift:114:21: error: argument for generic parameter 'S' could not be inferred
    - WARN  |  Swiftz/Swiftz/Chan.swift:66:7: warning: 'write' was deprecated in * version 2.1: Concurrency primitives are being moved to Concurrent.framework
    - WARN  |  Swiftz/Swiftz/Chan.swift:72:14: warning: 'read()' was deprecated in * version 2.1: Concurrency primitives are being moved to Concurrent.framework

Analyzed 1 podspec.

[!] The spec did not pass validation.
```

@ikesyo Thanks. Added it and got: 

```bash
➜  Swiftz git:(cocoapod) ✗ pod spec lint

 -> Swiftz (0.2.1)
    - ERROR | Returned an unsuccessful exit code. You can use `--verbose` for more information.
    - ERROR |  Swiftz/Swiftz/Comonad.swift:23:11: error: use of undeclared type 'Box'
    - ERROR |  Swiftz/Swiftz/Comonad.swift:34:11: error: use of undeclared type 'Box'
    - ERROR |  Swiftz/Swiftz/Comonad.swift:40:11: error: use of undeclared type 'Box'
    - ERROR |  Swiftz/Swiftz/Pointed.swift:15:11: error: use of undeclared type 'Box'
    - ERROR |  Swiftz/Swiftz/Pointed.swift:21:11: error: use of undeclared type 'Result'
    - ERROR |  Swiftz/Swiftz/Pointed.swift:29:11: error: use of undeclared type 'Either'
    - ERROR |  Swiftz/Swiftz/Those.swift:12:12: error: use of undeclared type 'Box'
    - ERROR |  Swiftz/Swiftz/Those.swift:13:12: error: use of undeclared type 'Box'
    - ERROR |  Swiftz/Swiftz/Those.swift:14:13: error: use of undeclared type 'Box'
    - ERROR |  Swiftz/Swiftz/NonEmptyList.swift:13:20: error: use of undeclared type 'Box'
    - ERROR |  Swiftz/Swiftz/ArrayExt.swift:312:9: error: cannot invoke 'span' with an argument list of type '([A], (_) -> _)'
    - NOTE  |  Swiftz/Swiftz/ArrayExt.swift:312:13: note: expected an argument list of type '([A], (A -> Bool))'
    - ERROR |  Swiftz/Swiftz/ArrayZipper.swift:46:25: error: binary operator '<^>' cannot be applied to operands of type 'A -> B' and '[A]'
    - ERROR |  Swiftz/Swiftz/Arrow.swift:135:22: error: use of undeclared type 'Either'
    - ERROR |  Swiftz/Swiftz/Arrow.swift:137:23: error: use of undeclared type 'Either'
    - ERROR |  Swiftz/Swiftz/Arrow.swift:140:23: error: use of undeclared type 'Either'
    - ERROR |  Swiftz/Swiftz/Arrow.swift:145:23: error: use of undeclared type 'Either'
    - ERROR |  Swiftz/Swiftz/Bifunctor.swift:44:19: error: use of undeclared type 'Either'
    - ERROR |  Swiftz/Swiftz/Bifunctor.swift:42:16: error: use of undeclared type 'Either'
    - ERROR |  Swiftz/Swiftz/Bifunctor.swift:48:57: error: use of undeclared type 'Either'
    - ERROR |  Swiftz/Swiftz/Bifunctor.swift:57:40: error: use of undeclared type 'Either'
    - ERROR |  Swiftz/Swiftz/Bifunctor.swift:61:41: error: use of undeclared type 'Either'
    - ERROR |  Swiftz/Swiftz/Bifunctor.swift:38:8: error: type 'EitherBF<A, C>' does not conform to protocol 'Bifunctor'
    - ERROR |  Swiftz/Swiftz/Bifunctor.swift:49:10: error: 'EitherBF<A, C>' does not have a member named 'e'
    - ERROR |  Swiftz/Swiftz/Bifunctor.swift:50:9: error: enum case pattern cannot match values of the non-enum type '<<error type>>'
    - ERROR |  Swiftz/Swiftz/Bifunctor.swift:52:9: error: enum case pattern cannot match values of the non-enum type '<<error type>>'
    - ERROR |  Swiftz/Swiftz/Bifunctor.swift:110:24: error: argument for generic parameter 'S' could not be inferred
    - NOTE  |  Swiftz/Swiftz/Iso.swift:35:13: note: in call to function 'identity'
    - ERROR |  Swiftz/Swiftz/Bifunctor.swift:114:21: error: argument for generic parameter 'S' could not be inferred
    - WARN  |  Swiftz/Swiftz/Chan.swift:66:7: warning: 'write' was deprecated in * version 2.1: Concurrency primitives are being moved to Concurrent.framework
    - WARN  |  Swiftz/Swiftz/Chan.swift:72:14: warning: 'read()' was deprecated in * version 2.1: Concurrency primitives are being moved to Concurrent.framework

Analyzed 1 podspec.

[!] The spec did not pass validation.
```

--- a/Swiftz.podspec
+++ b/Swiftz.podspec
@@ -1,0 +1,15 @@
+Pod::Spec.new do |s|
+  s.name        = "Swiftz"
+  s.version     = "0.2.1"
+  s.summary     = "Swiftz is a Swift library for functional programming."
+  s.homepage    = "https://github.com/typelift/Swiftz"
+  s.license     = { :type => "BSD" }
+  s.authors     = { "CodaFi" => "devteam.codafi@gmail.com", "pthariensflame" => "alexanderaltman@me.com" }
+
+  s.requires_arc = true
+  s.osx.deployment_target = "10.9"
+  s.ios.deployment_target = "8.0"
+  s.source   = { :git => "https://github.com/typelift/Swiftz.git", :tag => "v#{s.version}", :submodules => true }
+  s.source_files = "Swiftz/*.swift", "**/Swiftx/*.swift"
+  s.exclude_files = "**/Swiftx/Operators.swift"
+end

